research_project/greedy_triangulation.py

```diff
@@ -9,16 +9,32 @@
 
 from src.functions import poipairs_by_distance, new_edge_intersects
 
+def create_pois_groups(subgraph_percentages: list[float], pois: list[int]) -> list[int]:
+    pois_groups = []
+    pois_not_added = pois
+    for i in range (0, len(pois)):
+        group = random.sample(pois_not_added, int(len(pois) * subgraph_percentages[i]))
```

Comment:
`pois` and `subgraph_percentages` does not have the same length. We should just iterate over percentages

Also, using `int` here poses the same problem we talked about at the university. Rounding down repeatedly means we can get a significant amount of remaining vertices.

```suggestion
    for p in subgraph_percentages:
        group = random.sample(pois_not_added, int(len(pois) * p))
```